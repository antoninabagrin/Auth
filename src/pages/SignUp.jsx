import React from 'react';
import {
  FormControlLabel,
  TextField,
  Box,
  Button,
  Avatar,
  Checkbox,
  Grid,
  Typography,
  Link,
} from '@mui/material';
import { useNavigate } from 'react-router-dom';
import axios from '../utils/axios';

export function SignUp() {
  const history = useNavigate();
  const res = (allData) => {
    axios
      .post('/auth/signup', {
        firstName: allData.firstName,
        lastName: allData.lastName,
        email: allData.email,
        password: allData.password,
        username: allData.username,
      })
      .then(function (response) {
        if (response.data === 'USER_CREATED' && response.status === 201) {
          console.log(history);
          history('/signin');
        }
      })
      .catch(function (error) {
        console.log(error);
      });
  };
  const handleSubmit = (e) => {
    e.preventDefault();

    const data = new FormData(e.currentTarget);
    const allData = {
      email: data.get('email'),
      lastName: data.get('lastName'),
      firstName: data.get('firstName'),
      password: data.get('password'),
      passwordConform: data.get('passwordConfirm'),
      username: data.get('username'),
    };
    if (
      allData.email &&
      allData.firstName &&
      allData.lastName &&
      allData.password &&
      allData.passwordConform &&
      allData.username
    ) {
      if (allData.password === allData.passwordConform) {
        res(allData);
      } else {
        alert('Password and confirm password doesnt match');
      }
    } else {
      alert('You should fill every field ');
    }
  };

  return (
    <>
      <Box
        sx={{
          marginTop: 8,
          display: 'flex',
          flexDirection: 'column',
          alignItems: 'center',
        }}
      >
        <Avatar></Avatar>
        <Typography component='h1' variant='h5' margin='5'>
          Sign up
        </Typography>
        <Box component='form' onSubmit={handleSubmit}>
          <Grid container spacing={2}>
            <Grid item xs={12} sm={6}>
              <TextField
                autoComplete='given-name'
                name='firstName'
                required
                fullWidth
                id='firstName'
                label='First Name'
                autoFocus
              />
            </Grid>
            <Grid item xs={12} sm={6}>
              <TextField
                required
                fullWidth
                id='lastName'
                label='Last Name'
                name='lastName'
                autoComplete='family-name'
              />
            </Grid>
            <Grid item xs={12}>
              <TextField
                required
                fullWidth
                id='email'
                label='Email Address'
                name='email'
                autoComplete='email'
              />
            </Grid>
            <Grid item xs={12}>
              <TextField
                required
                fullWidth
                name='password'
                label='Password'
                type='password'
                id='password'
                autoComplete='new-password'
              />
            </Grid>
            <Grid item xs={12}>
              <FormControlLabel
                control={<Checkbox value='term' color='primary' />}
                label=' I accept the terms and conditions.'
              />
            </Grid>
          </Grid>
          <Button type='submit' fullWidth variant='contained'>
            Join
          </Button>
          <Grid container justifyContent='flex-end'>
            <Link href='#' variant='body2'>
              <Grid item>Already have an account? Sign in</Grid>
            </Link>
          </Grid>
        </Box>
      </Box>
    </>
  );
}
