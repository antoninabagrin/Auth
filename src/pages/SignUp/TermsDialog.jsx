import * as React from 'react';
import {
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogContentText,
  DialogTitle,
  Divider,
  IconButton,
} from '@mui/material';
import { useTranslation } from 'react-i18next';
import FileDownloadIcon from '@mui/icons-material/FileDownload';
import FileSaver from 'file-saver';

export default function TermsDialog({ open, setOpen, setFieldValue }) {
  const { t } = useTranslation();

  const handleDownload = (event) => {
    event.preventDefault();
    const terms = new Blob(
      [
        t(
          'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Suscipit quocorporis cupiditate voluptates sapiente exercitationem doloremque quibusdam assumenda iste! Id fuga earum saepe eligendi labore maiores nesciunt ipsam iusto sequi.',
        ),
      ],
      { type: 'text/plain;charset=utf-8' },
    );
    FileSaver.saveAs(terms, 'Terms.txt');
  };

  return (
    <Dialog open={open} onClose={() => setOpen(false)}>
      <DialogTitle>{t('Terms and Conditions')}</DialogTitle>
      <Divider />
      <DialogContent>
        <DialogContentText>
          {t(
            'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Suscipit quocorporis cupiditate voluptates sapiente exercitationem doloremque quibusdam assumenda iste! Id fuga earum saepe eligendi labore maiores nesciunt ipsam iusto sequi.',
          )}
        </DialogContentText>
        <IconButton onClick={handleDownload}>
          <FileDownloadIcon></FileDownloadIcon>
        </IconButton>
      </DialogContent>
      <Divider />
      <DialogActions>
        <Button
          onClick={() => {
            setOpen(false);
            setFieldValue('acceptTerms', false);
          }}
        >
          {t('Cancel')}
        </Button>
        <Button
          onClick={() => {
            setFieldValue('acceptTerms', true);
            setOpen(false);
          }}
          autoFocus
        >
          {t('Agree')}
        </Button>
      </DialogActions>
    </Dialog>
  );
}
