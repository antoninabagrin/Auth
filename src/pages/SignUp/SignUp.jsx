import React from 'react';
import { useFormik } from 'formik';
import * as Yup from 'yup';
import TermsDialog from './TermsDialog';
import axios from '../../utils/axios';
import { useNavigate } from 'react-router-dom';
import Box from '@mui/material/Box';
import {
  Avatar,
  Button,
  Checkbox,
  Container,
  FormControlLabel,
  Grid,
  Link,
  TextField,
  Typography,
} from '@mui/material';
import { useTranslation } from 'react-i18next';

export default function SignUp() {
  const [open, setOpen] = React.useState(false);
  const [isAgree, setIsAgree] = React.useState(false);
  const { t } = useTranslation();
  const history = useNavigate();
  const errorStyle = { color: 'red' };

  const formik = useFormik({
    initialValues: {
      email: '',
      username: '',
      password: '',
      confirmPassword: '',
      acceptTerms: false,
    },
    validationSchema: Yup.object({
      email: Yup.string().email(t('Invalid email')).required(t('Required')),
      username: Yup.string()
        .max(10, t('Must be 10 characters or less'))
        .required(t('Required')),
      password: Yup.string().required(t('Password is required')),
      confirmPassword: Yup.string().oneOf(
        [Yup.ref('password'), null],
        t('Passwords must match'),
      ),
    }),
    onSubmit: (values) => {
      console.log('values', values);
      axios
        .post('/auth/signup', {
          email: values.email,
          password: values.password,
          confirmPassword: values.confirmPassword,
          username: values.username,
        })
        .then(function (response) {
          if (response.data === 'USER_CREATED' && response.status === 201) {
            history('/signin');
          }
        })
        .catch(function (error) {
          console.log(error);
        });
    },
  });

  const checkDisableJoinButton = () => {
    return !isAgree;
  };

  return (
    <Grid container direction='row' justifyContent='center' alignItems='center'>
      <TermsDialog open={open} setOpen={setOpen} setIsAgree={setIsAgree} />
      <Container maxWidth='sm'>
        <Box
          sx={{
            margin: 8,
            display: 'flex',
            flexDirection: 'column',
            alignItems: 'center',
          }}
        >
          <Avatar />
          <Typography component='h1' variant='h5'>
            {t('Sign up')}
          </Typography>
          <Box component='form' onSubmit={formik.handleSubmit}>
            <TextField
              margin='normal'
              fullWidth
              autoFocus
              label={t('Email Address')}
              id='email'
              name='email'
              onChange={formik.handleChange}
              value={formik.values.email}
            />
            {formik.touched.email && formik.errors.email ? (
              <div style={errorStyle}>{formik.errors.email}</div>
            ) : null}
            <TextField
              margin='normal'
              fullWidth
              autoFocus
              label={t('Username')}
              id='username'
              name='username'
              onChange={formik.handleChange}
              value={formik.values.username}
            />
            {formik.touched.username && formik.errors.username ? (
              <div style={errorStyle}>{formik.errors.username}</div>
            ) : null}
            <TextField
              margin='normal'
              fullWidth
              autoFocus
              label={t('Password')}
              id='password'
              type='password'
              name='password'
              onChange={formik.handleChange}
              value={formik.values.password}
            />
            {formik.touched.password && formik.errors.password ? (
              <div style={errorStyle}>{formik.errors.password}</div>
            ) : null}
            <TextField
              margin='normal'
              fullWidth
              autoFocus
              label={t('Confirm password')}
              id='confirmPassword'
              type='password'
              name='confirmPassword'
              onChange={formik.handleChange}
              value={formik.values.confirmPassword}
            />
            {formik.touched.confirmPassword && formik.errors.confirmPassword ? (
              <div style={errorStyle}>{formik.errors.confirmPassword}</div>
            ) : null}
            <FormControlLabel
              control={
                <Checkbox
                  onClick={() => setOpen(true)}
                  value='term'
                  color='primary'
                />
              }
              label={t('I accept the terms and conditions.')}
            />
            {formik.touched.acceptTerms && formik.errors.acceptTerms ? (
              <div style={errorStyle}>{formik.errors.acceptTerms}</div>
            ) : null}

            <Button
              type='submit'
              fullWidth
              variant='contained'
              disabled={checkDisableJoinButton()}
            >
              {t('Join')}
            </Button>

            <Grid container justifyContent={'flex-end'}>
              <Link href='/signin' variant='body2'>
                {t('Already have an account? Sign in!')}
              </Link>
            </Grid>
          </Box>
        </Box>
      </Container>
    </Grid>
  );
}
