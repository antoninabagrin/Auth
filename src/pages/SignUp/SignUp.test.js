import { render, screen } from '@testing-library/react';
import SignUp from './SignUp';
import '@testing-library/jest-dom/extend-expect';

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key) => key }),
}));

const mockedUsedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUsedNavigate,
}));

test('check SignUp render', () => {
  render(<SignUp />);
  const joinButton = screen.getByTestId('joinButton');
  expect(joinButton).toBeDisabled();
});
