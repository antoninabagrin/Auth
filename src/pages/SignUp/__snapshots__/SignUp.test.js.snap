// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp SignUp render 1`] = `
<input
  aria-invalid="false"
  class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
  data-testid="required-confirmPassword"
  id="confirmPassword"
  name="confirmPassword"
  required=""
  type="password"
  value=""
/>
`;

exports[`SignUp SignUp render 2`] = `
<input
  aria-invalid="false"
  class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
  data-testid="required-confirmPassword"
  id="confirmPassword"
  name="confirmPassword"
  required=""
  type="password"
  value="Hey"
/>
`;
